### DISCUSSION TITLE
Option to restart the vpn connection after connection loss

### DISCUSSION THREAD

**Discussion by Lululoe** [haugene/docker-transmission-openvpn#2295] [Q&A] [answered]
I'm currently using the container on Truenas (Kubernetes) and PIA is my VPN Provider. My Internet connection isn't the most stable and will disconnect randomly (once every two days or so) and when this happens the container will stop downloading any active or new torrents. Would it be possible to either reconnect the vpn after the connection is back or restart the container after a certain amount of time?

**Comment by ilike2burnthing** [comment, marked as answer]
https://haugene.github.io/docker-transmission-openvpn/config-options/#health_check_option

Also use ENV `SELFHEAL`=`true`

**Comment by craftcm** [comment]
> I'm currently using the container on Truenas (Kubernetes) and PIA is my VPN Provider. My Internet connection isn't the most stable and will disconnect randomly (once every two days or so) and when this happens the container will stop downloading any active or new torrents. Would it be possible to either reconnect the vpn after the connection is back or restart the container after a certain amount of time?

Did you happen to follow a guide to get docker-transmission-openvpn installed on TrueNAS Scale? Since it doesn’t support Docker run or compose, I’ve been attempting to get it setup manually by adding a new Docker container through the TrueNAS GUI, but have had a lot of trouble. Thank you for any guidance you can offer.
